pkg/build/generator/generator.go

```diff
@@ -482,84 +478,89 @@ func (g *BuildGenerator) generateBuildFromConfig(ctx kapi.Context, bc *buildapi.
 		sourceImage.From.Kind = "DockerImage"
 		sourceImage.From.Name = sourceImageSpec
 		sourceImage.From.Namespace = ""
-		build.Spec.Source.Images[i] = sourceImage
+		Source.Images[i] = sourceImage
 	}
+	return nil
+}
 
-	// If the Build is using a From reference instead of a resolved image, we need to resolve that From
-	// reference to a valid image so we can run the build.  Builds do not consume ImageStream references,
-	// only image specs.
+// setBuildStrategy set strategy item for new builds
+func (g *BuildGenerator) setBuildStrategy(ctx kapi.Context, builderSecrets []kapi.Secret, bcCopy *buildapi.BuildConfig, strategy *buildapi.BuildStrategy) error {
```

Comment:
this isn't setting the strategy, it's setting the base image (and possibly the associated pull secret) for the build's strategy.